### PR TITLE
Feature/app test

### PR DESCRIPTION
**Pull request by ADavidson02**
#### What does this PR do?
* Moves the Movies component to tdb as we decide if we need this component or if it will be removed
* Change the Route path from Movies
* Move Component did mount into App component 
* Add an integration test for App component 

#### Where should the reviewer start?
* In the App test file

#### How should this be manually tested?
* Run the App test with npm test

#### Any background context you want to provide?
* Some information regarding Movies is still in the file but commented out

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,40 +1,61 @@
...
         <Switch>
           <Route
             path="/movies/:movie_id"
-            render={props => (
+            render={(props) => (
               <MovieDetails data={this.state.selectedMovie} {...props} />
             )}
           />
-          <Route
+          <React.Fragment>
```
Comment (brabbuss) [inline src/App.js:38]: I will dig into why we were getting the test error without surrounding this in <React.Fragment>, but seeing as we get no errors in console or in tests, this seems like a good solution for the time being

File: src/App.test.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+
+    const movieCard = await waitFor(() =>
+      screen.getByRole("heading", { name: /mulan 49%/i })
+    );
+    expect(movieCard).toBeInTheDocument();
```
Comment (brabbuss) [inline src/App.test.js:22]: I forgot - I believe there is a way to determine how many occurences there are as well. That might be a good assertion to add to this unit test

File: src/App.test.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    render(
+      <MemoryRouter>
+        <App />
+      </MemoryRouter>
+    );
+
+    userEvent.click(
+      await waitFor(() => screen.getByRole("link", { name: /337401 poster/i }))
+    );
+    expect(
```
Comment (brabbuss) [inline src/App.test.js:54]: I think we can also add an `expect` that determines whether or not the elements that should no longer be on the page are indeed gone (`MovieCard`)

File: src/components/MovieCard.jsx [hunk 4]
```diff
@@ -17,7 +17,7 @@ const MovieCard = props => {
         style={{ flex: "13rem", cursor: "pointer", border: "none" }}
         className="card mt-3"
         onClick={() => props.getMovieDetails(id)}>
-        <Link to={`/movies/${id}`}><img className="card-img-top" src={poster} alt="" /></Link>
+        <Link to={`/movies/${id}`}><img className="card-img-top" src={poster} alt={`${id} poster`} /></Link>
```
Comment (brabbuss) [inline src/components/MovieCard.jsx:20]: I like this a lot better than by targeting an empty string. :-D

File: src/components/tbd/Movies.test.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { render, screen, waitFor } from "@testing-library/react";
```
Comment (brabbuss) [inline src/components/tbd/Movies.test.js:1]: I think we can actually junk this test since we migrated functionality into `App`

**Review by brabbuss — APPROVED** [review]
Everything looks good. I think there's room for a few more `expect` statements in the TDD. I'm glad we migrated the functionality of `Movies` into `App`. Feels cleaner and less confusing.

File: src/App.js (hunk 1 above)
Comment (letakeane) [reply src/App.js:38]: Generally we only use React Fragments to wrap JSX elements when no other HTML element is needed. You don't typically see it INSIDE another element.

File: src/App.js [hunk 6]
```diff
@@ -1,40 +1,61 @@
...
+              {movies.map((movie) => (
+                <MovieCard
+                  key={movie.id}
+                  data={movie}
+                  getMovieDetails={this.getMovieDetails}
+                />
+              ))}
+            </div>
+          </React.Fragment>
+          {/* <Route
```
Comment (letakeane) [inline src/App.js:49]: Instead of commenting out, simply delete. You can recreate the code :) Trust your skills!

Comment (letakeane) [reply src/App.js:49]: Additionally, commented out code should not be left in production code.

File: src/App.test.js (hunk 3 above)
Comment (letakeane) [reply src/App.test.js:54]: There is in fact a query that returns `null` if nothing is found.

File: src/components/MovieCard.test.js [hunk 7]
```diff
@@ -1,47 +1,48 @@
...
-      expect(mockGetMovieDetails).toHaveBeenCalledWith(fakeMovie.id)
-
-    
-  })
-
-
-})
\ No newline at end of file
+        <MovieCard
+          key={fakeMovieData.id}
```
Comment (letakeane) [inline src/components/MovieCard.test.js:17]: Thanks for fixing the indentation! Looks so much more readable now :heart:

File: src/components/MovieCard.test.js [hunk 8]
```diff
@@ -1,47 +1,48 @@
...
+          <MovieCard
+            key={fakeMovie.id}
+            data={fakeMovie}
+            getMovieDetails={mockGetMovieDetails}
+          />
+        </BrowserRouter>
+      );
+
+       userEvent.click(screen.getByRole("link", { name: /694919 poster/i }));
+      expect(mockGetMovieDetails).toHaveBeenCalledWith(fakeMovie.id);
```
Comment (letakeane) [inline src/components/MovieCard.test.js:46]: Perfection! :tada:

File: src/components/MovieDetails.jsx [hunk 9]
```diff
@@ -1,6 +1,7 @@
 import React from "react";
 import "../css/MovieDetails.scss";
 
+
```
Comment (letakeane) [inline src/components/MovieDetails.jsx:4]: You can get rid of this dead line 😄

File: src/components/MovieDetails.jsx [hunk 10]
```diff
@@ -30,10 +31,12 @@ const MovieDetails = props => {
         <p className="card-text">
           <small className="text-muted">Release Date {release_date}</small>
         </p>
+        {tagline && <p>{tagline}</p>}
```
Comment (letakeane) [inline src/components/MovieDetails.jsx:34]: Excellent conditional rendering! Great job paying attention to the fluctuating data!

File: src/components/MovieDetails.test.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+      tagline: "",
+      average_rating: 4.909090909090909,
+    };
+
+    render(<MovieDetails data={movieData} />);
+
+    expect(screen.getByText("Mulan 49%")).toBeInTheDocument();
+    expect(screen.getByText("Release Date 2020-09-04")).toBeInTheDocument();
+    expect(screen.getByRole('img')).toBeInTheDocument();
+  });
```
Comment (letakeane) [inline src/components/MovieDetails.test.js:31]: Don't forget to add tests to capture the conditional rendering you added in MovieDetails! Give MovieDetails a movie without/without various properties and make sure it's rendering as expected.

**Pull request merged**